src/main/java/com/redhat/cloud/common/clowder/configsource/ClowderConfigSource.java

```diff
@@ -143,9 +143,13 @@ public Set<String> getPropertyNames() {
 
         if (exposeKafkaSslConfigKeys) {
             for (String key : KAFKA_SASL_KEYS) {
-                String value = getValue(key);
-                if (value != null && !value.isBlank()) {
-                    availableProperties.add(key);
+                try {
+                    String value = getValue(key);
+                    if (value != null && !value.isBlank()) {
+                        availableProperties.add(key);
+                    }
+                } catch (IllegalStateException ie) {
+                    log.debugf(ie.getMessage());
```

Comment:
Was this a temporary addition or is it meant to be merged?

Reply:
Yes it have to be merge.
Because to build property name collection, we have to check if the property exists or not, to add only valued data.
The `getValue` method will parse the config file value to check if it exists.
Nevertheless, regarding other configuration data, some values could be inconsistent and rise an `IllegalStateException`.
For instence, a `kafka.sasl.jaas.config` property is inconsistent on a simple Kafka SSL config case, `getValue` method should rise an exception and this key should not be present on returned set of `getPropertyNames`.

Reply:
Make sense. Do we want to log as `debug` level though? Also, logging `getMessage()` prevents the stack trace from being fully logged.

Reply:
Thats the idea, having a single message and not a full stack trace because it will be not helpfull in this case.

Reply:
```suggestion
                } catch (IllegalStateException ie) {
                    log.debug(ie.getMessage());
```

Reply:
What about my question about the log level? 😄

Reply:
Because this catch block can be reach on a regular case (nor only in case of error), it should be logged only if you are seaching for somthing unexpected, to investiguate on such thing, we usualy update log level to debug, this is why I selected this level.
What do you have in mind ?

Reply:
Based on your answer, I'm okay with `debug` now 😃